route.js

```diff
@@ -17,15 +17,66 @@ app.listen(port, (err) => {
 
 
 app.get('/', (request, response) => {
-    response.send('Bienvenue sur Express');
+  response.send('Bienvenue sur Express');
 });
   
 app.get('/users', (req, res) => {
-    connection.query('SELECT * from DVM_Legal_Entity', (err, results) =>{
-        if(err) {
-            res.status(500).send('No body was here..')
-        } else {
-            res.json(results);
-        }
-    })
-});
\ No newline at end of file
+  connection.query('SELECT * from DVM_Legal_Entity', (err, results) =>{
+    if(err) {
+      res.status(500).send('No body was here..')
+    }
+    res.json(results);
+  })
+});
+
+app.get('/users/:id', (req, res) => {
+  const { id } = req.params
+  connection.query('SELECT * from DVM_Legal_Entity WHERE id = ?', id, (err, results) => {
+    if(err) {
+      res.status(500).send('Internal server error')
+    }
+    if (results.length === 0) {
+      res.status(404).send('User ID not found')
+    } else {
+      res.json(results)
+    }
+  })
+})
+
+app.post('/users', (req, res) => {
+  const formData = req.body;
+  connection.query('INSERT INTO DVM_Legal_Entity SET ?', formData, (err, results) => {
+    if (err.code){
```

Comment:
if there is no error, err is undefined: this will crash

the 400 status should handled before querying database. take a look here: https://github.com/WildCodeSchool/reims-js-202003-pjt-tonton-sommelier/blob/master/server/server.js#L24